### PR TITLE
Add instructions for packaged builds

### PR DESCRIPTION
**Pull request by lmbr-pip**
Early draft of package build documentation

* Working on Linux instructions currently, will add those in second PR.

### CODE REVIEW

File: Documentation/PackedAssetBuilds.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+### Reprocess MultiplayerSample_Dependencies.xml
+
+> Step only required if  https://github.com/o3de/o3de/issues/15338 is still open.
+
+* Run AssetProcessor 
+* Open the Assets tab
+* Open the Product Assets tab
+* Search for `multiplayersample_dependencies.xml`, right-click on it, and choose "Reprocess Source Asset"
+* Close AssetProcessor
+
```
Comment (mbalfour-amzn) [inline Documentation/PackedAssetBuilds.md:51]: This section can be removed now.

File: Documentation/PackedAssetBuilds.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+Run the  AssetBundler
+
+```
+build\windows\bin\profile\AssetBundler.exe --project-path="c:\your\path\to\o3de-multiplayersample"
+```
+
+Follow steps for "Create a bundle for game assets", "Create a bundle for engine assets" and "Add bundles to the release game layout" from https://www.o3de.org/docs/user-guide/packaging/asset-bundler/bundle-assets-for-release/
+
+* The "default seed lists" choice should choose all but 4 seed lists to make the `engine_pc.pak`
+* The other 4 seed lists should all get selected to make the `game_pc.pak`
```
Comment (mbalfour-amzn) [inline Documentation/PackedAssetBuilds.md:78]: This should probably list exactly what "the other 4 seed lists" are.

File: Documentation/PackedAssetBuilds.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+mkdir %1\Gems\AWSCore
+ 
+rem Copy the pak files
+copy c:\your\path\to\o3de-multiplayersample\AssetBundling\Bundles\*.pak %1\Cache\pc
+ 
+rem Copy the executables and DLLs
+copy c:\your\path\to\o3de-multiplayersample\build\windows_mono\bin\profile\*.* %1
+ 
+rem Copy the AWSCore files?
+copy c:\your\path\to\o3de-multiplayersample\build\windows_mono\bin\profile\Gems\AWSCore\*.* %1\Gems\AWSCore
```
Comment (mbalfour-amzn) [inline Documentation/PackedAssetBuilds.md:100]:
```suggestion
rem Copy the AWSCore files
copy c:\your\path\to\o3de-multiplayersample\build\windows_mono\bin\profile\Gems\AWSCore\*.* %1\Gems\AWSCore
```

File: Documentation/PackedAssetBuilds.md [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+ 
+rem Copy the launch_client / launch_server files
+copy c:\your\path\to\o3de-multiplayersample\launch_*.* %1
+ 
+rem Copy the PIX DLLs if PIX is enabled in the build
+:: copy D:\ly\lyengine\3rdParty\WinPixEventRuntime\bin\x64 %1
```
Comment (mbalfour-amzn) [inline Documentation/PackedAssetBuilds.md:106]: This should say \your\path\to\3rdParty or something

File: Documentation/PackedAssetBuilds.md [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+```
+
+After running, check the output logs to verify there aren't any crashes, missing assets, etc. If any assets are missing, go back to the AssetBundler step, add the assets, and repeat.
+
+### Run the Launchers from Visual Studio
+Rename the o3de-multiplayersample/Cache/pc folder to pc-loose
+Create an o3de-multiplayersample/Cache/pc folder
+Copy the engine_pc.pak and game_pc.pak files into that folder
+Use the same command-line parameters as above
+
```
Comment (mbalfour-amzn) [inline Documentation/PackedAssetBuilds.md:128]: This should probably be removed. It's definitely not required for building the pak builds, and these instructions are a little messy and incomplete for describing how to run the pak builds from VS.

File: Documentation/PackedAssetBuilds.md [hunk 6]
```diff
@@ -0,0 +1,122 @@
+# Packaged MultiplayerSample Builds
+To make relocatable client and server builds for the MultiplayerSample, we recommend making packaged builds. These package builds will contain the Game or Server Launcher and the bundled assets needed to run the launcher outside of the developer environment.
+
+You can make both release packaged builds or profile packaged builds, see https://www.o3de.org/docs/user-guide/packaging/windows-release-builds/.
```
Comment (allisaurus) [inline Documentation/PackedAssetBuilds.md:4]:
```suggestion
You can make both release packaged builds or profile packaged builds. For more information about creating release builds, see the O3DE documentation on [Creating a Project Game Release Layout for Windows](https://www.o3de.org/docs/user-guide/packaging/windows-release-builds/).
```

File: Documentation/PackedAssetBuilds.md [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+
+The guide below covers how to make profile packaged builds which are very useful for early sharing and play testing.
+
+## Windows Profile PAK Setup
+
+
```
Comment (allisaurus) [inline Documentation/PackedAssetBuilds.md:10]: nit: extra line break

File: Documentation/PackedAssetBuilds.md [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+```shell
+cmake --build build\windows --target AssetBundler --config profile -- /m /nologo
+```
+
+### Build monolithic game
+
+Build a second version of the executables as monolithic pak builds.
+
+``` shell
+// Create build files for a monolithic build that also disables all user/project registry settings overrides
```
Comment (allisaurus) [inline Documentation/PackedAssetBuilds.md:47]:
```suggestion
# Create build files for a monolithic build that also disables all user/project registry settings overrides
```
using `#` comments in a `shell` code block leads to better syntax highlighting:

image

File: Documentation/PackedAssetBuilds.md [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+cmake -B build\windows_mono -S . -G "Visual Studio 16" -DLY_3RDPARTY_PATH=c:\your\path\to\3rdParty -DLY_MONOLITHIC_GAME=1 -DALLOW_SETTINGS_REGISTRY_DEVELOPMENT_OVERRIDES=0
+
+// Build the profile versions of all the executables
```
Comment (allisaurus) [inline Documentation/PackedAssetBuilds.md:50]:
```suggestion
# Build the profile versions of all the executables
```

File: Documentation/PackedAssetBuilds.md [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+cmake --build build\windows_mono --target MultiplayerSample.GameLauncher MultiplayerSample.ServerLauncher MultiplayerSample.UnifiedLauncher --config profile -- /m /nologo
+```
+
+The outputs in windows_mono\bin\profile can be copied and run anywhere, once the pak files are put in the proper location.
+
+
+### Bundle Content
+
+Run the  AssetBundler
```
Comment (allisaurus) [inline Documentation/PackedAssetBuilds.md:59]:
```suggestion
Run the AssetBundler
```

**Review by allisaurus — APPROVED** [review]

File: Documentation/PackedAssetBuilds.md [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+mkdir %1
+mkdir %1\Cache
+mkdir %1\Cache\pc
+mkdir %1\Gems
+mkdir %1\Gems\AWSCore
+ 
+rem Copy the pak files
+copy c:\your\path\to\o3de-multiplayersample\AssetBundling\Bundles\*.pak %1\Cache\pc
+ 
+rem Copy the executables and DLLs
```
Comment (AMZN-Gene) [inline Documentation/PackedAssetBuilds.md:86]: The *.* will also copy in gigs of debug symbol pdb’s might call out removing them by hand or update the script

**Review by AMZN-Gene — APPROVED** [review]

File: Documentation/PackedAssetBuilds.md (hunk 11 above)
Comment (lmbr-pip) [reply Documentation/PackedAssetBuilds.md:86]: Good call out, for now will add a note

File: Documentation/PackedAssetBuilds.md [hunk 12]
```diff
@@ -0,0 +1,122 @@
+# Packaged MultiplayerSample Builds
+To make relocatable client and server builds for the MultiplayerSample, we recommend making packaged builds. These package builds will contain the Game or Server Launcher and the bundled assets needed to run the launcher outside of the developer environment.
+
+You can make both release packaged builds or profile packaged builds. For more information about creating release builds, see the O3DE documentation on [Creating a Project Game Release Layout for Windows](https://www.o3de.org/docs/user-guide/packaging/windows-release-builds/).
+
+The guide below covers how to make profile packaged builds which are very useful for early sharing and play testing.
+
+## Windows Profile PAK Setup
+
```
Comment (lmbr-pip) [inline Documentation/PackedAssetBuilds.md:9]:
```suggestion
```

File: Documentation/PackedAssetBuilds.md [hunk 13]
```diff
@@ -0,0 +1,122 @@
...
+copy c:\your\path\to\o3de-multiplayersample\build\windows_mono\bin\profile\Gems\AWSCore\*.* %1\Gems\AWSCore
+ 
+rem Copy the launch_client / launch_server files
+copy c:\your\path\to\o3de-multiplayersample\launch_*.* %1
+ 
+rem Copy the PIX DLLs if PIX is enabled in the build
+:: copy c:\your\path\to\3rdParty\WinPixEventRuntime\bin\x64 %1
+```
+
+It's recommended during testing you track the commit IDs used from o3de, o3de-multiplayersample, and o3de-multiplayersample-assets to have as reference.
```
Comment (lmbr-pip) [inline Documentation/PackedAssetBuilds.md:99]:
```suggestion
Note: The script above will copy everything from the profile directory. You can either remove files you don't require such as .pdb files to save space, or modify the script to copy just whats needed.

It's recommended during testing that you include source information for your build, such as the commit IDs for o3de, o3de-multiplayersample, and o3de-multiplayersample-assets.
```

**Pull request merged**